ci/publish-crate.sh

```diff
@@ -26,14 +26,27 @@ expectedCrateVersion="$MAJOR.$MINOR.$PATCH$SPECIAL"
   exit 1
 }
 
+# check workspace.version for worksapce root
+workspace_cargo_tomls=(Cargo.toml programs/sbf/Cargo.toml)
+for cargo_toml in "${workspace_cargo_tomls[@]}"; do
+  if ! grep -q "^version = \"$expectedCrateVersion\"$" "$cargo_toml"; then
+    echo "Error: Cargo.toml version is not $expectedCrateVersion"
+    exit 1
+  fi
+done
+
 Cargo_tomls=$(ci/order-crates-for-publishing.py)
 
 for Cargo_toml in $Cargo_tomls; do
   echo "--- $Cargo_toml"
-  grep -q "^version = \"$expectedCrateVersion\"$" "$Cargo_toml" || {
-    echo "Error: $Cargo_toml version is not $expectedCrateVersion"
-    exit 1
-  }
+
+  # check the version which doesn't inherit from worksapce
+  if ! grep -q "^version = { workspace = true }$" "$Cargo_toml"; then
+    grep -q "^version = \"$expectedCrateVersion\"$" "$Cargo_toml" || {
```

Comment:
What do you think about either logging a warning here, or skipping the `$expectedCrateVersion` check and fail as soon as the `workspace = true` check fails. That way if we add more workspace manifests this will remind us to add them to line 30 above.

Reply:
yeah. you're right. I think we need a warning between them.

Reply:
fixed: [8ac90cc](https://github.com/solana-labs/solana/pull/31894/commits/8ac90cc2bbaf152314e66656f5014c944654ddd1)